### PR TITLE
AMP-47972 JRE Kotlin example

### PR DESCRIPTION
**Pull request by falconandy**

### CODE REVIEW

File: jre/kotlin/AmpliApp/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Ampli JRE SDK (Kotlin)
+An example app using the Ampli JRE SDK
+
+# Usage
+
+### Setup the project
+You will need to do the following before running the app.
+
+1. Create a `.env` with your API key
+    1. `cp .env.example .env`
```
Comment (justin-fiedler) [inline jre/kotlin/AmpliApp/README.md:10]: Nice, using .env here is great. Can update the Android examples to match?

**Review by justin-fiedler — CHANGES_REQUESTED** [review]
Overall looks great. See minor nit for more event tracking calls.

File: jre/kotlin/AmpliApp/README.md (hunk 1 above)
Comment (falconandy) [reply jre/kotlin/AmpliApp/README.md:10]: @justin-fiedler It seems it's possible in some way - https://github.com/cdimascio/dotenv-kotlin#android-usage. Should we use it? Maybe it's not simpler than the current `secrets.xml`

File: jre/kotlin/AmpliApp/src/main/kotlin/Main.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    ampli.identify(userId, Identify(
+        requiredNumber = 42.0,
+    ))
+
+    val extra = MiddlewareExtra(mapOf("extra-key" to "extra-value"))
+    ampli.track(userId, EventWithOptionalProperties(
+        optionalBoolean = true,
+    ), extra = extra)
+
+    ampli.eventWithAllProperties(userId, EventWithAllProperties(
```
[the remark this answers is not in the document]
Comment (falconandy) [reply jre/kotlin/AmpliApp/src/main/kotlin/Main.kt:97]: Added

File: jre/kotlin/AmpliApp/README.md (hunk 1 above)
Comment (justin-fiedler) [reply jre/kotlin/AmpliApp/README.md:10]: Ok thanks, let's keep the current secrets.xml

**Review by justin-fiedler — APPROVED** [review]

**Pull request merged**